extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ArcRecorder.java

```diff
@@ -55,13 +66,29 @@ public BeanContainer initBeanContainer(ArcContainer container, List<BeanContaine
         return beanContainer;
     }
 
-    public void handleLifecycleEvents(ShutdownContext context, BeanContainer beanContainer) {
-        LifecycleEventRunner instance = beanContainer.instance(LifecycleEventRunner.class);
-        instance.fireStartupEvent();
+    public void handleLifecycleEvents(ShutdownContext context, LaunchMode launchMode,
+            boolean disableApplicationLifecycleObservers) {
+        ArcContainerImpl container = ArcContainerImpl.instance();
+        List<Class<?>> mockBeanClasses = new ArrayList<>();
```

Comment:
Super minor nitpick: Perhaps we can avoid the allocation here this list will rarely be populated?

Reply:
Done.